src/raftstore/coprocessor/split_check/keys.rs

```diff
@@ -171,26 +176,35 @@ impl<C: CasualRouter + Send> SplitCheckObserver for KeysCheckObserver<C> {
     }
 }
 
-/// Get the approximate number of keys in the range.
-pub fn get_region_approximate_keys(db: &DB, region: &Region) -> Result<u64> {
+/// Get approximate number of keys in the range. Return result of CF_WRITE derectly if it's not empty.
+/// For RawKV, return result of CF_DEFAULT.
+pub fn get_region_approximate_keys(db: &DB, region: &Region) -> Result<(CfName, u64)> {
     // try to get from RangeProperties first.
-    match get_region_approximate_keys_cf(db, CF_WRITE, region) {
-        Ok(v) => {
-            if v > 0 {
-                return Ok(v);
+    let cfs = vec![CF_WRITE, CF_DEFAULT];
+    for cf in cfs {
+        let mut num_keys_cf = match get_region_approximate_keys_cf(db, cf, region) {
+            Ok(v) => v,
+            Err(e) => {
+                debug!(
+                    "failed to get keys from RangeProperties";
+                    "err" => ?e,
+                );
+                0
             }
+        };
+        if num_keys_cf == 0 {
```

Comment:
you can move this part to `Err(e)`

Reply:
@Connor1996 No we need to do this if Ok(v) return 0

Reply:
why?